portafly/src/components/data-list/reducers/BulkActionsReducer.ts

```diff
@@ -19,12 +19,21 @@ const defaultBulkActions: BulkActionsState = {
 }
 
 // Action Handlers
+const SET_MODAL = 'SET_MODAL'
+const CLOSE_MODAL = 'CLOSE_MODAL'
+const ACTION_START = 'ACTION_START'
+const ACTION_SUCCESS = 'ACTION_SUCCESS'
+const ACTION_FAILED = 'ACTION_FAILED'
+
+type SetModalAction = Action<BulkAction>
+type ActionFailedAction = Action<string>
+
 const BulkActionsActionHandlers: ActionHandlers<BulkActionsState, any> = {
-  SET_MODAL: (state, action: Action<BulkAction>) => ({ ...state, modal: action.payload }),
-  CLOSE_MODAL: (state) => ({ ...state, modal: undefined, errorMsg: undefined }),
-  ACTION_START: (state) => ({ ...state, isLoading: true }),
-  ACTION_SUCCESS: () => defaultBulkActions,
-  ACTION_FAILED: (state, action: Action<string>) => (
+  [SET_MODAL]: (state, action: SetModalAction) => ({ ...state, modal: action.payload }),
```

Comment:
BulkAction || string || undefined... we need to review this in the future...

Reply:
Agreed